pkg/services/ngalert/state/state.go

```diff
@@ -112,6 +113,19 @@ func (a *State) resultNoData(alertRule *ngModels.AlertRule, result eval.Result)
 	return a
 }
 
+func (a *State) NeedsSending(resendDelay time.Duration) bool {
+	if a.State != eval.Alerting {
+		return false
+	}
+	if a.EndsAt.After(a.LastSent) {
```

Comment:
wouldn't this always bypass the delay? A running alert should always end after it was last sent - unless, I'm not understanding this correctly.

Reply:
This logic was copied from the prom NeedsSending but I think you're right. The difference is that we're not tracking the resolved at in the state. Makes me think that we ought to.